setup.py

```diff
@@ -136,6 +139,18 @@
 ENABLE_DOCUMENTATION_BUILD = os.environ.get(
     'GRPC_PYTHON_ENABLE_DOCUMENTATION_BUILD', False)
 
+def check_linker_need_libatomic():
+  """Test if linker on system needs libatomic.
+  """
+  code_test = (b'#include <atomic>\n' +
```

Comment:
I don't think this is guaranteed to be an lock-free on all platforms. Only `std::atomic_flag` gives that guarantee. See 29.7.2 of the C++ specification.

Reply:
This isn't related with C++ standard but with how gcc and libatomic are implemented. GCC makes atomic operation library as a platform specific extension. ([ref](https://gcc.gnu.org/wiki/Atomic/GCCMM) This code tries to use `__atomic_load_8` which is one of functions from `libatomic` and 64 bit value is chosen because GCC can generate code for values with a size less than 32 bits without `libatomic`.

Reply:
Thank you for the review.`libatomic` is defined by gcc and `libstdc++` uses gcc atomic extensions so it's fair to assume that it eventually calls it.